src/controller/debtor-controller.ts

```diff
@@ -0,0 +1,233 @@
+/**
+ *  SudoSOS back-end API service.
+ *  Copyright (C) 2020  Study association GEWIS
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published
+ *  by the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import BaseController, { BaseControllerOptions } from './base-controller';
+import { Response } from 'express';
+import log4js, { Logger } from 'log4js';
+import Policy from './policy';
+import { RequestWithToken } from '../middleware/token-middleware';
+import { parseRequestPagination } from '../helpers/pagination';
+import DebtorService from '../service/debtor-service';
+import User from '../entity/user/user';
+import { asArrayOfUserTypes, asDate } from '../helpers/validators';
+import { In } from 'typeorm';
+import { HandoutFinesRequest } from './request/debtor-request';
+import Fine from '../entity/fine/fine';
+
+export default class DebtorController extends BaseController {
+  private logger: Logger = log4js.getLogger(' DebtorController');
+
+  public constructor(options: BaseControllerOptions) {
+    super(options);
+    this.logger.level = process.env.LOG_LEVEL;
+  }
+
+  public getPolicy(): Policy {
+    return {
+      '/': {
+        GET: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'get', 'all', 'Fine', ['*']),
+          handler: this.returnAllFineHandoutEvents.bind(this),
+        },
+      },
+      '/:id(\\d+)': {
+        GET: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'get', 'all', 'Fine', ['*']),
+          handler: this.returnSingleFineHandoutEvent.bind(this),
+        },
+      },
+      '/single/:id(\\d+)': {
+        DELETE: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'delete', 'all', 'Fine', ['*']),
+          handler: this.deleteFine.bind(this),
+        },
+      },
+      '/eligible': {
+        GET: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'get', 'all', 'Fine', ['*']),
+          handler: this.calculateFines.bind(this),
+        },
+      },
+      '/handout': {
+        POST: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'get', 'all', 'Fine', ['*']),
+          handler: this.handoutFines.bind(this),
+        },
+      },
+    };
+  }
+
+  /**
+   * Get all fine handout events
```

Comment:
mist unieke @operationId

Reply:
Resolved in cfcbd191b85b3707a4b3e2321c59065dc4a95ea6